grabs locations from geocode api by city name

Riley-Strandlund opened the pull request:
This module only queries the API by city names that exist in the United States.

If you search by state it doesn't give all of the cities/counties in the state it only returns the state.

There is an if statement that removes the 3rd index of a location result which is intended to only remove zip codes or other numbers that pop up inconsistently in that index. Most queries return 4 things so anymore than that and the 3rd index is removed.

Riley-Strandlund commented:
The location  data grabbed includes [[city, county, state, country], [latitude, longitude]]. Should we exclude county? It can serve as a unique location identifier if somehow city, county, and state are identical.

Jon-117 commented:
@Riley-Strandlund yeah, I agree that county is not necessary and may cause issues with the search. Take it out 👍

Riley-Strandlund commented:
Added line of code to get rid of the index containing county names. Issue #7

Jon-117 commented:
Looks good mostly, but there are a few things to address. One is writing the file as if it's going to run directly. The other is a possibility we need to account for: multiple towns being returned in the data. Try searching Minneapolis as an example. 

At the end, the module needs to return a new Location class object too. See #20. We won't actually be running anything from this file directly so main() is potentially confusing function name to call from the main.py/app.py file. 

We're only using one and it needs to be right, so the user should be displayed results and asked to choose. This type of interaction will have to be repeated with both Location and the Playlist types at least... Am I missing any other places a user will need to be presented with a choice? 

What do you think is the best option for handling this?

Riley-Strandlund commented:
I changed the name for the method that was called 'main' and changed it to 'get_location'. If someone wants to get a location they only need to call 'get_location' and add user input to get locations.

Almost all of the locations a user can enter into the system will have more than one place being returned. We want the user to choose from all of the returned items to increase the chances of finding the location they want.

The places the user is presented a choice are where they select location, favorite weather mood, recent (to find and favorite playlists), and the weather playlist list (Shows the list of playlists based on location). So I think we need 4 total.

Lastly, are you asking how we will display the compiled data to the user?

Jon-117 commented:
> Lastly, are you asking how we will display the compiled data to the user?

I guess I was wondering if this should be a function of the UI program. IE, present list to user, ask for user selection, return user selection.

Jon-117 commented on get_location.py:
```diff
@@ -0,0 +1,53 @@
+    userInput = userInput
+    searchString = f'{userInput}'
+    params.update({'city': searchString})
+    pass
```
`pass` is unnecessary here. `pass` is usually placeholder for doing nothing, but the function already does stuff.

Riley-Strandlund replied:
I agree it was a placeholder, but I forgot to get rid of it. There is a new version of the file that resolves this issue.

Jon-117 commented on get_location.py:
```diff
@@ -0,0 +1,53 @@
+        latitude = info['lat'] # Latitude and longitude can be used for openweather to locate the area.
+        longitude = info['lon']
+        print(city, state, longitude, latitude)
+        locationObj = Location(city, state, latitude, longitude)
```
You'd had the variables in line with PEP 8 naming conventions before, but changed to camel in the latest commit. Variables like `locationObj` and `userInput` should use snake_case (`location_obj`and `user_input`).  

Not a big deal I think, so we can implement for now and come back to refactor later if time permits.

Riley-Strandlund replied:
I checked the file over and everything is in snake case (the underscore naming convention).

Jon-117 commented on get_location.py:
```diff
@@ -0,0 +1,53 @@
+"""Grab location from API based on users input. Currently set up to search for city names based on input only in the United States."""
+""" Currently when looking up by state only returns the state instead of all cities in the state"""
+import requests
+
```
add `from classes import Location`

looks like classes file wasn't updated before you started the file. Will merge. Should pull latest updates, then branch and fix.

Jon-117 reviewed (approved):
Will merge. Minor changes necessary, but merging allows to update classes file for proper import.

Jon-117 commented on get_location.py:
```diff
@@ -0,0 +1,53 @@
+    try:
+        create_params(userInput)
+        returned_data = request_nominatim()
+        format_data(returned_data)
```
You're formatting the data but haven't returned it.

try:
```
def get_location(userInput):
    """Gather all methods into one in order to create location object. 
    Only needs to call this function to excecute all of them."""
    try:
        create_params(userInput)
        returned_data = request_nominatim()
        return(format_data(returned_data))
```

I got overzealous in approving the file. I apologize.

Riley-Strandlund replied:
The file returns 'actual_cities' now so it can give the UI the location options for the user. The location object will also be made in main because it would be too complicated/weird returning something then reusing it. Unless you have a good way of doing it.
